shaders/src/light_indirect.fs

```diff
@@ -203,21 +194,122 @@ float computeSpecularAO(float NoV, float ao, float roughness) {
  * - The reflected vector may be modified to point towards the dominant specular
  *   direction to match reference renderings when the roughness increases
  */
-vec3 getReflectedVector(const PixelParams pixel, const vec3 n) {
+
+vec3 getReflectedVector(const vec3 v, const PixelParams pixel, const vec3 n) {
 #if defined(MATERIAL_HAS_ANISOTROPY)
     vec3  anisotropyDirection = pixel.anisotropy >= 0.0 ? pixel.anisotropicB : pixel.anisotropicT;
-    vec3  anisotropicTangent  = cross(anisotropyDirection, shading_view);
+    vec3  anisotropicTangent  = cross(anisotropyDirection, v);
     vec3  anisotropicNormal   = cross(anisotropicTangent, anisotropyDirection);
     float bendFactor          = abs(pixel.anisotropy) * saturate(5.0 * pixel.roughness);
     vec3  bentNormal          = normalize(mix(n, anisotropicNormal, bendFactor));
 
-    vec3 r = reflect(-shading_view, bentNormal);
+    vec3 r = reflect(-v, bentNormal);
+#else
+    vec3 r = reflect(-v, n);
+#endif
+    return r;
+}
+
+vec3 getReflectedVector(const PixelParams pixel, const vec3 n) {
+#if defined(MATERIAL_HAS_ANISOTROPY)
+    vec3 r = getReflectedVector(shading_view, pixel, n);
 #else
     vec3 r = shading_reflected;
 #endif
     return getSpecularDominantDirection(n, r, pixel.linearRoughness);
 }
 
+//------------------------------------------------------------------------------
+// Prefiltered importance sampling
+//------------------------------------------------------------------------------
+
+#if IBL_INTEGRATION == IBL_INTEGRATION_IMPORTANCE_SAMPLING
+vec3 importanceSampleIBL(vec3 v, float NoV, const PixelParams pixel) {
```

Comment:
Could you move this to another file? It adds a lot of code we don't really need to see most of the time

Reply:
It would be kind-of annoying because this function needs getReflectedVector(), but is needed by evaluateIBL(), so it needs to be defined between those two. Moving it to another file, we'd also need to move getReflectedVector().

Reply:
Ok we can keep it like this for now. But pass `PixelParams` as the first parameter to match the other functions